rest-rest/src/Components/MimeTypes.tsx

```diff
@@ -0,0 +1,131 @@
+import { Select, SelectOption, SelectVariant } from '@patternfly/react-core';
+import React from 'react';
+
+//Code strongly "inspired" from https://www.patternfly.org/v4/components/select/#typeahead
+
+export interface IMimeTypeSelectInputForm {
+  label: string;
+  onChange: any;
+  value: string;
+  values: string[];
+}
+
+export interface IMimeTypeStatus {
+  isOpen: boolean;
+  selected: string | undefined;
+  label: string;
+  values: string[];
+  newOptions: JSX.Element[];
+}
+
+class MimeTypeSelectInput extends React.Component<IMimeTypeSelectInputForm, IMimeTypeStatus> {
+  onToggle: (isOpen: any) => void;
+  onSelect: (event: any, selection: any, isPlaceholder: any) => void;
+  onCreateOption: (newValue: any) => void;
+  constructor(props: IMimeTypeSelectInputForm) {
+    super(props);
+
+    this.state = {
+      isOpen: false,
+      selected: props.value,
+      label: props.label,
+      values: props.values,
+      newOptions: []
+    };
+
+    this.onToggle = isOpen => {
+      this.setState({
+        isOpen
+      });
+    };
+
+    // @ts-ignore
+    this.onSelect = (event, selection, isPlaceholder) => {
+      if (isPlaceholder) this.clearSelection();
+      else {
+        this.setState({
+          selected: selection,
+          isOpen: false
+        });
+      }
+      props.onChange(selection);
+    };
+
+    this.onCreateOption = newValue => {
+      this.setState({
+        newOptions: [...this.state.newOptions, <SelectOption key={newValue} value={newValue} />]
+      });
+    };
+
+    this.clearSelection = () => {
+      this.setState({
+        selected: undefined,
+        isOpen: false
+      });
+    };
+  }
+  clearSelection() {
+    throw new Error('Method not implemented.');
+  }
+
+  render() {
+    const { isOpen, selected, label, values } = this.state;
+    let mimetypes: string[] = values;
+
+    if (!mimetypes || mimetypes.length == 0) {
+      //List of mimetypes from https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types  
+      mimetypes = ["audio/aac", "application/x-abiword", "application/x-freearc", "image/avif", "video/x-msvideo", "application/vnd.amazon.ebook",
```

Comment:
Could we please move this array definition into a separate file, otherwise we create a new array on every render.